Pset_UndergroundExcavationCommon

larswik opened the pull request:
# Pset: Pset_UndergroundExcavationCommon

## Applicability: IfcUndergroundExcavation

## Description: 

Common properties for underground excavation elements

| Property                    | Datatype                          | Description                                                  |
| --------------------------- | --------------------------------- | ------------------------------------------------------------ |
| UndergroundExcavationMethod | PEnum_UndergroundExcavationMethod | Method of excavation                                         |
| ExcavationMethodOther       | IfcLabel                          | A user defined underground excavation method when UndergroundExcavationMethod = USERDEFINED |
| SupportDescription          | IfcLabel                          | Description of Support                                       |
| SupportClass       | IfcLabel                          | Specifies a class defining the support measures for underground excavation.  |
| ExcavationClass    | IfcLabel                          | Specifies a class defining the support measures for underground excavation.  |
| RoundLength                 | IfcPositiveLengthMeasure          | Length of individual round or drive                          |

## Enumeration: PEnum_UndergroundExcavationMethod

| Constant                     | Description |
| ---------------------------- | ----------- |
| HAND                         | Manual excavation method using hand tools.  |
| EXCAVATOR                    | Mechanized method utilizing excavator machinery for efficient large-scale underground earthmoving.  |
| JACKHAMMER                   | Pneumatic or hydraulic percussion tool breaking rock or concrete.  |
| RIPPER                       | Equipment with a toothed tool to break and loosen hard soil.  |
| DRILLANDBLAST                | A method of desintegrating rock by drilling small diameter holes on a planned layout, packing these with explosives and then firing to a fixed program to shatter the rock in a desired form.  |
| ROADHEADER                   | Machine with rotating cutter head specifically designed for the excavation of roadways, tunnels, and underground chambers.  |
| TBMOPENWITHGRIPPERS          | An open rock tunnel boring machine which generally utilizes roller disc cutters as excavation tools and which moves forward by reacting  against the tunnel walls through a hydraulic gripper reaction system. |
| TBMCLOSEDWITHGRIPPERS          | A closed-face shield which generally utilizes roller disc cutters as excavation tools and which moves forward by reacting  against the tunnel walls through a hydraulic gripper reaction system. |
| TBMEARTHPRESSUREFACESUPPORT  | A closed-face shield designed for tunnelling in fine-grained soils by trapping excavated materials against the face and removing them at a rate slow enough to maintain pressures that counterbalance earth pressures exerted against the cutterhead and prevent ingress of water; which moves forward by reacting  against an installed lining. |
| TBMSLURRYPRESSUREFACESUPPORT | A closed-face shield designed for tunnelling in very soft, wet, or running ground by use of circulating, pressurized clay slurry against the face to counterbalance earth pressures exerted against the cutterhead, prevent ingress of water and also to carry away the excavated material; which moves forward by reacting  against an installed lining. |
| TBMAIRPRESSUREFACESUPPORT    | A closed-face shield designed for tunnelling in unstable soils with the presence of water where compressed air is used to counterbalance earth pressures exerted against the cutterhead; which moves forward by reacting against an installed lining. |
| TBMMIXSHIELD                 | A closed-face shield  with a mixed face support system that combines various methods to counterbalance earth pressures exerted against the cutterhead; which moves forward by reacting against an installed lining. |
| SHAFTBORINGMACHINE           | Specialized machine for boring vertical or inclined shafts.  |
| RAISEDRILL                   | Type of shaft boring machine that enlarges a pilot hole by reaming to the full shaft diameter from the bottom to the top of the shaft. |
| USERDEFINED                        |             |

larswik committed:
Pset_UndergroundExcavationCommon

larswik committed:
Update Documentation.md

larswik committed:
Update Documentation.md

larswik committed:
Update Documentation.md

larswik committed:
Update Documentation.md

larswik committed:
Update Documentation.md

larswik committed:
Update Documentation.md

larswik committed:
Update Documentation.md

SergejMuhic commented on IFC4x3/Properties/e/ExcavationClassification_0M$dI8P5T3I8$lTN_LlVyH/DocProperty.xml:
```diff
@@ -0,0 +1,2 @@
+﻿<?xml version="1.0" encoding="utf-8"?>
```
Should this not be modelled with [Classification Association](https://bsi-infraroom.github.io/IFC-Documentation-Tunnel/4_4_0_0/general/HTML/link/classification-association.htm)?

larswik replied:
I agree, but involving @JontyDunn in this conversation.

larswik replied:
Rename to ExcavationClass. @JontyDunn or @hessf-ilf might improve the description, e.g. with an example. Right now it is "Specifies an excavation class used in underground excavation."

SergejMuhic commented on IFC4x3/Properties/r/RoundLength_2Px7GAEKr9oPOVi4hIhBDX/Documentation.md:
```diff
@@ -0,0 +1 @@
+Length of individual round or drive.
```
What is a "Round"?

larswik replied:
Ping @JontyDunn.

larswik replied:
Ping @hessf-ilf

SergejMuhic commented on IFC4x3/Properties/s/SupportClassification_09LzP6dcn6OAdINBkUxEqS/DocProperty.xml:
```diff
@@ -0,0 +1,2 @@
+﻿<?xml version="1.0" encoding="utf-8"?>
```
Also, Classification Association?

larswik replied:
I agree, but involving @JontyDunn in this conversation.

larswik replied:
Rename to SupportClass. @JontyDunn or @hessf-ilf might improve the description, e.g. with an example. Right now it is "Specifies a class defining the support measures for underground excavation."

SergejMuhic reviewed (commented):
Constants and enums do not have documentation.

larswik committed:
Merge remote-tracking branch 'upstream/tunnel' into feature/Pset_UndergroundExcavationCommon

larswik committed:
Updating according to comments.

Also removing OTHER and adding (re-using) USERDEFINED for PEnum_UndergroundExcavationMethod.

larswik commented:
In PEnum_UndergroundExcavationMethod, I removed (and deleted) OTHER and added (re-used) USERDEFINED instead.

larswik committed:
PEnum_UndergroundExcavationMethod constants descriptions.

larswik commented:
> Constants and enums do not have documentation.

Fixed in [d48ad31](https://github.com/bSI-InfraRoom/IFC-Specification/pull/570/commits/d48ad310dc37278bb55272f1a387eb3c23823d09)

SergejMuhic commented on IFC4x3/Properties/e/ExcavationMethodOther_1fa7O2cGn6_fZVIehnm$Jw/Documentation.md:
```diff
@@ -0,0 +1 @@
+A user defined underground excavation method when UndergroundExcavationMethod = USERDEFINED.
```
Property name says Other.

larswik replied:
> Property name says Other.

I have no good proposal on what to call this and make it short?

larswik replied:
We decided to remove this.

SergejMuhic commented on IFC4x3/PropertyEnumerations/u/PEnum_UndergroundExcavationMethod.xml:
```diff
@@ -0,0 +1,21 @@
+<?xml version="1.0" encoding="utf-8"?>
```
Does not have a description.

SergejMuhic commented on IFC4x3/PropertyConstants/c/COREDRILLING_3kAMQZBc19h9RiBVqacIVy.xml:
```diff
@@ -1,5 +1,5 @@
 <?xml version="1.0" encoding="utf-8"?>
-<DocPropertyConstant xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" id="COREDRILLING_3kAMQZBc19h9RiBVqacIVy" Name="COREDRILLING" UniqueId="ee2966a3-2e60-49ac-96ec-2dfd249927fc">
```
Apparently removed.

SergejMuhic commented on IFC4x3/PropertyConstants/d/DESTRUCTIVEDRILLING_0kW0UHgtH7WPGblh83KWIC.xml:
```diff
@@ -1,5 +1,5 @@
 <?xml version="1.0" encoding="utf-8"?>
-<DocPropertyConstant xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" id="DESTRUCTIVEDRILLING_0kW0UHgtH7WPGblh83KWIC" Name="DESTRUCTIVEDRILLING" UniqueId="2e800791-ab74-4781-9425-beb20352048c">
```
Seems removed also.

SergejMuhic commented on IFC4x3/PropertyConstants/t/TBMCLOSEDWITHGRIPPERS_24tst5kj171RquDTEFq43J.xml:
```diff
@@ -0,0 +1,5 @@
+<?xml version="1.0" encoding="utf-8"?>
+<DocPropertyConstant xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" id="TBMCLOSEDWITHGRIPPERS_24tst5kj171RquDTEFq43J" Name="TBMCLOSEDWITHGRIPPERS" UniqueId="84df6dc5-bad0-4705-bd38-35d38fd040d3">
+	<Documentation>A closed-face shield which generally utilizes roller disc cutters as excavation tools and which moves forward by reacting  against the tunnel walls through a hydraulic gripper reaction system.</Documentation>
```
```suggestion
	<Documentation>A closed-face shield which generally utilizes roller disc cutters as excavation tools and which moves forward by reacting against the tunnel walls through a hydraulic gripper reaction system.</Documentation>
```

larswik committed:
Update IFC4x3/PropertyConstants/t/TBMCLOSEDWITHGRIPPERS_24tst5kj171RquDTEFq43J.xml

Co-authored-by: SergejMuhic <sergejs1@gmail.com>

larswik committed:
Merge remote-tracking branch 'upstream/tunnel' into feature/Pset_UndergroundExcavationCommon

larswik committed:
Fixed according to comments/agreements

larswik commented:
Should be ok now. Removed the ExcavationMethodOther property.
